lib/node_modules/@stdlib/math/base/special/rising-factorial/src/main.c

```diff
@@ -0,0 +1,114 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*
+* ## Notice
+*
+* The original C++ code and copyright notice are from the [Boost library]{@link http://www.boost.org/doc/libs/1_64_0/boost/math/special_functions/factorials.hpp}. The implementation has been modified according to project conventions.
+*
+* ```text
+* (C) Copyright John Maddock 2006, 2010.
+*
+* Use, modification and distribution are subject to the
+* Boost Software License, Version 1.0. (See accompanying file
+* LICENSE or copy at http://www.boost.org/LICENSE_1_0.txt)
+* ```
+*/
+
+#include "stdlib/math/base/assert/is_integer.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/gamma_delta_ratio.h"
+#include "stdlib/math/base/special/falling_factorial.h"
+
+/**
+* Computes the rising factorial of `x` and `n`.
+*
+* ## Notes
+*
+* -   The rising factorial is defined as
+*
+*     ```tex
+*     \operatorname{stdlib_base_rising_factorial}(x, n) = x (x-1) (x-2) (x-3) \ldots (x-n+1)
+*     ```
+*
+*     or equivalently
+*
+*     ```tex
+*     \operatorname{stdlib_base_rising_factorial}(x, n) = \frac{ \Gamma(x + n) }{ \Gamma(x) };
+*     ```
+*
+* @param {number} x - first function parameter
+* @param {integer} n - second function parameter
+* @returns {number} function value
+*
+* @example
+* double v = stdlib_base_rising_factorial( 0.9, 5 );
+* // returns ~94.766
+*
+* @example
+* double v = stdlib_base_rising_factorial( -9.0, 3 );
+* // returns -504.0
+*
+* @example
+* double v = stdlib_base_rising_factorial( 0.0, 2 );
+* // returns 0.0
+*
+* @example
+* double v = stdlib_base_rising_factorial( 3.0, -2 );
+* // returns 0.5
+*/
+double stdlib_base_rising_factorial( const double x, const int32_t n ) {
+	double result;
+	int32_t nc;
+	double xc;
+	bool inv;
+
+	if ( stdlib_base_is_nan( x ) || !stdlib_base_is_integer( n ) ) {
```

Comment:
```suggestion
	if ( stdlib_base_is_nan( x ) ) {
```

`n` is already and integer.